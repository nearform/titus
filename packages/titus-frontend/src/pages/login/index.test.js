import React from 'react'
import {
  render,
  renderWithRouter,
  cleanup,
  waitFor,
  waitForElementToBeRemoved,
  fireEvent
} from '../../test-utils'
import Login from './index'

describe('<Login />', () => {
  afterEach(cleanup)

  it('renders without crashing', () => {
    const { getByLabelText, getByText } = render(<Login />)
    expect(getByLabelText(/username/i))
    expect(getByLabelText(/password/i))
    expect(getByText(/login/i))
  })

  it('shows required field errors', async () => {
    const { getByText } = render(<Login />)

    fireEvent.click(getByText(/login/i))
    await waitFor(() => {
      expect(getByText(/username is required/i)).toBeInTheDocument()
      expect(getByText(/password is required/i)).toBeInTheDocument()
    })
  })

  it('shows password requirements when password is inadequate', async () => {
    const { getByText, getByLabelText } = render(<Login />)

    const password = getByLabelText('Password:')
    fireEvent.change(password, { target: { value: 'Bar' } })
    fireEvent.click(getByText(/login/i))
    let error
    await waitFor(() => {
      error = getByText(/password must be/i)
    })
    fireEvent.change(password, { target: { value: 'Bar6' } })
    fireEvent.click(getByText(/login/i))
    await waitForElementToBeRemoved(error)
  })

  it('Login is successful', async () => {
    // Render with routing to detect url change
    const { getByText, getByLabelText, queryByText } = renderWithRouter(
      <Login />
    )
    await waitFor(() => {
      getByText(/login/i)
    })

    fireEvent.change(getByLabelText('Username:'), { target: { value: 'Foo' } })
    fireEvent.change(getByLabelText('Password:'), { target: { value: 'Bar6' } })
    expect(queryByText(/username is required/i)).not.toBeInTheDocument()
    expect(queryByText(/password is required/i)).not.toBeInTheDocument()

    fireEvent.click(getByText(/login/i))
    await waitFor(() => {
      getByText(/logout/i)
    })
  })
})
