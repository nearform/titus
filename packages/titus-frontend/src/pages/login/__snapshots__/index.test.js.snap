// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders without crashing 1`] = `
<DocumentFragment>
  <img
    alt="Titus logo"
    class="login-logo"
    src="img/Accel_Logo_Titus.svg"
  />
  <form
    novalidate=""
  >
    <header />
    <label
      for="username"
    >
      Username:
    </label>
    <input
      name="username"
      placeholder="Username"
      required=""
      type="text"
      value=""
    />
    <label
      for="password"
    >
      Password:
    </label>
    <input
      name="password"
      placeholder="Password"
      required=""
      type="password"
      value=""
    />
    <button
      class="button"
      type="submit"
    >
      Login
    </button>
    <sub />
  </form>
</DocumentFragment>
`;
